### ISSUE TITLE
Add an autosave project feature to QGIS

### ISSUE THREAD

**Issue by arno974** [qgis/QGIS#52753]
### Feature description

I know this subject has already been discussed (ticket https://github.com/qgis/QGIS/issues/24707), but has received little feedback.

Indeed, it seems to me interesting (IMHO even necessary) to have by default in QGIS an option to save the project automatically. There is of course the AutoSaver extension, but such functionality as autosave should be directly in QGIS instead of an extension. I know that it requires some code maintenance. Nevertheless, the logic of the AutoSaver code is about 50 lines and it is relatively simple code. This Auto Save option could be defined for example in the project properties (General > General settings). 

In my opinion, this would be an important improvement without requiring significant development.

I would be curious to have your opinion and feedback on this subject.

**Comment by WhitespaceQ** [comment]
If this is to be implemented, I think a simple versioning by file-name is a nice addition.

```
1. some-project.qgz

[auto-save, inside a subfolder]
2. some-project-timestamp.qgz
3. some-project-timestamp2.qgz

[manual-save]
4. replace current file to some-project.qgz

retain all auto-saved qgz
```
The timestamped file is due to not all backup/versioning software properly push the changed file within reasonable time before the same file is saved again.

with the same behavior for layers?

**Comment by baswein** [comment]
To me seems like a vital and important feature to have included in core QGIS. Before the qgz project file format there was a .qgs~ backup file that was always there and could potentially be resurrected in case of a crash which saved me a number of times. One thing I have noticed is that QGIS stops working during the save process. This lag is not necessarily a problem but feels very different than when you press the save button and you expect QGIS to stop responding and when the autosave suddenly does it for you. Solving this small detail would be on my wish list if this were implemented. Once again thank all of you for all of your work.

**Comment by Josefen** [comment]
I totally agree. Now that the autosaver plugin seems to be unsupported, the need is even greater. (The autosaver plugin hasnt  been updated in 26 month, and now it gererates a python error.)

**Comment by davidgilbertson** [comment]
+1, I just started using QGIS, and just had my first crash, and lost a bunch of stuff I'd done.

I'm so used to auto-save (literally no other piece of software I use requires manual saving) it's just something that I'm in the habit of doing.

Preferably we could get autosave on every change (if undo logic is robust), but even something like "every n minutes" would be better than nothing.

**Comment by Ipickedausername** [comment]
I don't understand the need to have an auto-saver feature which will only save the project. It does not create versions of the updated layers, so, by the end of the day, you have x version of the same data. 

In my opinion, (and use case) I would rather have an auto-save feature, that will create a temporary duplicate of all the layers (and not the project), in order to retrieve data if I notice something went wrong after saving the changes on a layer.

**Comment by arno974** [comment]
@Ipickedausername  in my opinion, it is two different needs. Iimagine that you have changed the symbology/label of a layer or created a map layout and QGIS crash, then you lost all what you did before.  An autosave feature would be useful in this case.

But, I agree having the possibility to save as well the editing layers would be useful. But in this case, it should probably be implemented in the editing toolbar.

**Comment by Ipickedausername** [comment]
Thank you for the explanations. 
Indeed, it would be great not to depends on plugins to save backup files. 

I don't know any user of QGIS around me who have never lost a huge amount of work because of crashes or wrong manipulations (the save button is terrible in these situations). 
After a while, you learn to always have backup layers.
